packages/language-server/src/plugins/typescript/TypeScriptPlugin.ts

```diff
@@ -324,6 +331,41 @@ export class TypeScriptPlugin
             .map(fix => mapCodeActionToParent(tsDoc, fix));
     }
 
+    onWatchFileChanges(fileName: string, changeType: FileChangeType) {
+        const scriptKind = getScriptKindFromFileName(fileName);
+
+        if (scriptKind === ts.ScriptKind.Unknown) {
+            return;
+        }
+
+        const tsconfigPath = findTsConfigPath(fileName);
+        const snapshotManager = SnapshotManager.getFromTsConfigPath(tsconfigPath);
+
+        if (changeType === FileChangeType.Deleted) {
+            snapshotManager.delete(fileName);
+            return
+        }
+
+        const content = ts.sys.readFile(fileName) ?? '';
+        const newSnapshot: DocumentSnapshot = {
+            ...ts.ScriptSnapshot.fromString(content),
+            // ensure it's greater than initial build
+            version: INITIAL_VERSION + 1,
+            scriptKind: getScriptKindFromFileName(fileName)
+        }
+
+        if (changeType === FileChangeType.Changed) {
```

Comment:
I just thought it shouldn't have any previous snapshot if the change type is created. But as you said it won't hurt. Maybe it would also force ts to reread the file if somehow a previous snapshot does exist.